clang_build/api.py

```diff
@@ -0,0 +1,20 @@
+from importlib import util as importlib_util
```

Comment:
Move to project. This has just one function that generates ... a project, just this time from a clang-build.py. The vague module name also hints at this function not needing a separate module.

Reply:
Currently cleaning up in a separate branch the "environment" class. I noticed that the "working_directory" is the folder of the top level `clang_build.toml`. So I think you should probably pass the path, and not the working directory to the `get_project` function of the loaded module.

Reply:
The signature is `get_project(directory, working_directory, environment, parent) -> Project`, where `directory` is the path where the Python file is, i.e. both are available.
IMO both can be useful in different cases.

Reply:
I don't see where the working_directory is needed. Also if you want the parent directory, you can just ask the parent. Naming it different things and having it appear multiple times is confusing I think.

Reply:
You are right - when there is a parent, the working_directory will be identical to `parent.directory` and if there is no parent, it will be identical to `directory`. This is seen nicely by the `test/py-api/subproject` test.
I will remove this unnecessary parameter from the API.